wordpress-mobile/WordPress-Android
# Issue/10370 auto upload error labels

Fixes #10370 

- Replaces `getNumberOfPostUploadErrorsOrCancellations` with `getNumberOfPostAutoUploadAttempts`. The `getNumberOfPostUploadErrorsOrCancellations` counter was used as a guard so we don't keep auto-uploading a post indefinitely. However, the counter was incremented whenever a media or post upload failed (even in offline). So if the user added 10 images in offline, the counter would be incremented ten times. Whereas `getNumberOfPostAutoUploadAttempts` is incremented only when the UploadStarter actually initiates the auto-upload. More info https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1366

- Decreases the number of auto-upload attempts to 3 (MAXIMUM_AUTO_UPLOAD_RETRIES)

- Updates the error messages when upload fails as desribed in #10370  - `Decision 2019-Aug-27`


To test:
1. Turn on airplane mode
2. Create a post
3. Add some media files
4. Click on publish
5. Notice "Post couldn't be published. We'll try again later" label is shown. Also notice both Cancel and Retry (under more) buttons are visible. The Label and Cancel button should be yellow.
------
1. Turn on airplane mode
2. Create a post
3. Add some media files
4. Change the post status to pending in Post settings and click on Submit
5. Notice "Post couldn't be submitted. We'll try again later" label is shown. Also notice both Cancel and Retry (under more) buttons are visible. The Label and Cancel button should be yellow.
6. Make the auto-upload fail 3 times (eg. turn the airplane mode off and on)
7. Notice after three attempts the message changes to "Couldn't perform operation. Post not submitted.". The cancel button should disappear and the label and Retry should be red.
------------
1. Create a post while online
2. Add some title and content (no media files)
3. Click on publish
4. Turn on airplane mode before the upload finishes
5. Notice "Post couldn't be published. We'll try again later" label is shown. Also notice both Cancel and Retry (under more) buttons are visible. The Label and Cancel button should be yellow.

Update release notes:

- [ x ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
